test/utils/image/manifest.go

```diff
@@ -137,6 +140,30 @@ var (
 	VolumeISCSIServer   = Config{e2eRegistry, "volume/iscsi", "1.0"}
 	VolumeGlusterServer = Config{e2eRegistry, "volume/gluster", "1.0"}
 	VolumeRBDServer     = Config{e2eRegistry, "volume/rbd", "1.0.1"}
+
+	// storage CSI GKE images
+	CSIGCPDsikCSIDriver = Config{gcrReleaseRegistry, "gcp-compute-persistent-disk-csi-driver", "v0.3.1-gke.0"}
```

Comment:
Keeping these definitions up-to-date will be a constant maintenance effort. Is it really necessary to define these `Config` instances? Eventually you will need to call `GetE2EImage` while patching the .yaml files, and at that time you have the current image string and can parse that to get registry, image name and tag.

Reply:
FYI once we have the CSI out of tree testing in good shape, we can work on removing the pd CSI encoding from the core e2e

Reply:
The effort should be the same if you maintain the current version in some YAML definition or in the `manifest.go` file.

So the main goal is to move all image definitions (registry, version) into the manifest.go file and then render the actual used yaml files like in: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/kubectl/kubectl.go#L265